Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/SpherePointer.cs

```diff
@@ -257,14 +268,81 @@ public bool TryUpdateQueryBufferForLayerMask(LayerMask layerMask, Vector3 pointe
 
                 for (int i = 0; i < numColliders; i++)
                 {
-                    grabbable = queryBuffer[i].GetComponent<NearInteractionGrabbable>();
+                    Collider collider = queryBuffer[i];
+                    grabbable = collider.GetComponent<NearInteractionGrabbable>();
+                    // Additional check: is grabbable in the camera frustrum
                     if (grabbable != null)
+                    {
+                        if (!isInFOV(collider))
+                        {
+                            grabbable = null;
+                        }
+                    }
+                    if (grabbable != null)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+
+            /// <summary>
+            /// Returns true if a collider's bounds is within the camera FOV
+            /// </summary>
+            /// <param name="myCollider">The collider to test</param>
+            /// <returns></returns>
+            private bool isInFOV(Collider myCollider)
+            {
+                Bounds b = myCollider.bounds;
+                var corners = new List<Vector3>();
+                BoundsExtensions.GetColliderBoundsPoints(myCollider, corners, 0);
+                float xMin = float.MaxValue, yMin = float.MaxValue, zMin = float.MaxValue;
+                float xMax = float.MinValue, yMax = float.MinValue, zMax = float.MinValue;
+                foreach (var pt in corners)
+                {
+                    if (isPointInFrustrum(pt))
                     {
                         return true;
                     }
+
+                    xMin = Mathf.Min(xMin, pt.x);
+                    yMin = Mathf.Min(yMin, pt.y);
+                    zMin = Mathf.Min(zMin, pt.z);
+                    xMax = Mathf.Max(xMax, pt.x);
+                    yMax = Mathf.Max(yMax, pt.y);
+                    zMax = Mathf.Max(zMax, pt.z);
+                }
+
+                // edge case: check if camera is inside the entire bounds of the collider;
+                var cameraPos = CameraCache.Main.transform.position;
+                if (xMin <= cameraPos.x && cameraPos.x <= xMax 
+                    && yMin <= cameraPos.y && cameraPos.y <= yMax
+                    && zMin <= cameraPos.z && cameraPos.z <= zMax)
+                {
+                    return true;
                 }
                 return false;
             }
+
+            /// <summary>
+            /// Returns true if a point is in the camera's frustrum
+            /// </summary>
+            /// <param name="pt">Point to test</param>
+            private bool isPointInFrustrum(Vector3 pt)
```

Comment:
Use CameraExtensions.IsInFOV here instead

Reply:
Oh cool, I didn't know about that. However, that function won't allow me to slightly increase the camera's FOV to have a slight buffer, or to use a distance from camera slightly larger or smaller. I could add a utility function that is a bit more flexible. I can explain in comments why I didn't use camera.IsInFov

Reply:
Use the Mathutilities is in fov then?

Reply:
Done